files/bootstrap.sh

```diff
@@ -137,6 +143,7 @@ PAUSE_CONTAINER_VERSION="${PAUSE_CONTAINER_VERSION:-3.1-eksbuild.1}"
 CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-dockerd}"
 IP_FAMILY="${IP_FAMILY:-}"
 SERVICE_IPV6_CIDR="${SERVICE_IPV6_CIDR:-}"
+ENABLE_LOCAL_OUTPOST="${ENABLE_LOCAL_OUTPOST:-}"
```

Comment:
I see this in the overview as well - `If --enable-local-outpost option is not set with true or false explicitly, bootstrap.sh will auto-detect whether the cluster is running in local outpost environment.`

Auto-discovery relies on making an EKS DescribeCluster call, and I don't want to do this for two reasons - 
* It forces a latency increase on every node launch. 
* It will be backwards incompatible for anyone running in a fully private VPC, since there's no private endpoints for EKS.

Why not default this to false? Local outposts isn't a common working configuration, so I'd expect this to run false for best backwards compatibility.

Reply:
@chunywan can you clarify here? I think @suket22 brings up a good point.

This DescribeCluster call only happens when `B64_CLUSTER_CA` or `API_SERVER_ENDPOINT` aren't defined, but AFAIK these will always be defined for non-custom AMI nodegroups. The only time they wouldn't be defined is for a custom AMI nodegroup, in which case the user could just use the `--enable-local-outpost` flag. Am I missing something? I admittedly don't have much context on Outpost.

Reply:
Regardless, this isn't changing when the DescribeCluster call happens, so I don't have any compatibility concerns. I'm just not sure the auto-detection mechanism is really useful.

Reply:
@suket22 Thanks for the comments. I think I need to refine the description of PR to make it clear.

> It forces a latency increase on every node launch.

This PR doesn't change whether the DescribeCluster is called or not. 
If it is called and `--enable-local-outpost` is not defined explicitly, the auto-discovery will be used. And the change is just add the filters to DescribeCluster call. 

> It will be backwards incompatible for anyone running in a fully private VPC, since there's no private endpoints for EKS.

As I mentioned above, if DescribeCluster is bypassed for private VPC (e.g., when B64_CLUSTER_CA or API_SERVER_ENDPOINT aren't defined) , this PR change doesn't call DescribeCluster either. 

So I think I will modify the description to make it clear.

Reply:
@suket22 @cartermckinnon 
I refined the description of the PR.